src/config_format/flb_cf_fluentbit.c

```diff
@@ -721,20 +721,18 @@ struct flb_cf *flb_cf_fluentbit_create(struct flb_cf *cf,
                                        char *file_path, char *buf, size_t size)
 {
     int ret;
-    int created = FLB_FALSE;
```

Comment:
I don't remember the exact reason but IIRC `created` was there because there are cases where the caller of this function can pass a `flb_cf` context and destroy it if it fails (maybe static config mode ?)

Reply:
> created was there because there are cases where the caller of this function can pass a flb_cf context and destroy it if it fails

I think so.

On the other hand, `flb_cf_yaml_create` doesn't check if `flb_cf` is argument or not.
The function always destroys `flb_cf` when some error occurs.
https://github.com/fluent/fluent-bit/blob/v1.9.7/src/config_format/flb_cf_yaml.c#L1095


If this patch will be merged, `flb_cf_fluentbit_create` also destroys when some error occurs to propagate error.

Reply:
got it make sense